src/Actions/MakeQueueTenantAwareAction.php

```diff
@@ -39,18 +38,15 @@ protected function listenForJobsBeingQueued(): self
     protected function listenForJobsBeingProcessed(): self
     {
         app('events')->listen(JobProcessing::class, function (JobProcessing $event) {
-            $tenantId = $event->job->payload()['tenantId'] ?? null;
-
-            if (! $tenantId) {
+            if (! array_key_exists('tenantId', $event->job->payload())) {
                 return;
             }
 
-            if (! config('multitenancy.queues_are_tenant_aware_by_default')) {
-                return;
-            }
+            $tenantId = $event->job->payload()['tenantId'];
+
             /** @var \Spatie\Multitenancy\Models\Tenant $tenant */
-            if (! $tenant = $this->getTenantModel()::find($tenantId)) {
-                return;
+            if (! $tenantId || ! ($tenant = $this->getTenantModel()::find($tenantId))) {
+                throw NoCurrentTenant::make();
```

Comment:
Instead of using this exception, let's create a dedicated exception for this situation. Might be called `NoCurrentTenantFoundInTenantAwareJob`